docs/Design/SoftDetailedDes/MIS.tex

```diff
@@ -306,6 +306,77 @@ \subsubsection{Local Functions}
 
 \newpage
 
+\section{MIS of Authentication Module} \label{mAuthentication} 
+The Authentication module acts to "log-in" the user by providing the client with a JWS token.
+
+\subsection{Module}
+
+authentication
+
+\subsection{Uses}
+
+ Database Driver Module (Section \ref{mDBDriver}) 
+
+\subsection{Syntax}
+
+\subsubsection{Exported Constants}
+None
+
+\subsubsection{Exported Access Programs}
+
+\begin{center}
+\begin{tabular}{p{5cm} p{3cm} p{3cm} p{5cm}}
+\hline
+\textbf{Name} & \textbf{In} & \textbf{Out} & \textbf{Exceptions} \\
+\hline
+getToken & String, String & String & invalidCredentialsException \\
+verifyToken & String & String & invalidTokenException \\
+\hline
+\end{tabular}
+\end{center}
+
+\subsection{Semantics}
+
+\subsubsection{State Variables}
+None
+
+\subsubsection{Environment Variables}
+PrivateKey (for JWT)
+PublicKey (for JWT)
+
+\subsubsection{Assumptions}
+None
+
+\subsubsection{Access Routine Semantics}
+
+\noindent getToken(\textit{email}, \textit{passwordEncrypted}):
+\begin{itemize}
+\item transition: none
```

Comment:
Can you add a transition where a signedToken is assigned using the PrivateKey? I know you have it in the output but I think it should also be mentioned in the transition, like "assigns user a signed token using the PrivateKey if credentials are correct".

Essentially, if we have environment variables being mentioned, you should put in the transition how the interaction with the environment variables work. If there are other places that utilize the environment variables or interact with it, maybe add a transition there too. For example, if PublicKey is assigned anywhere or used anywhere, maybe include a transition for it if possible as well.